Bad CPU type in executable failure in loading Native library in Apple Silicon with M4 Pro

### Describe the bug

Tried 

`make` 

Or 

`make release PROFILES="-Pspark-3.5 -Pscala-2.13"`

> [ERROR] Failed to execute goal com.github.os72:protoc-jar-maven-plugin:3.11.4:run (default) on project comet-spark-spark3.5_2.13: Unable to execute protoc-jar for /Users/prashantsingh/Desktop/oss-workspace/arrow-datafusion-comet/native/proto/src/proto/partitioning.proto: Cannot run program "/Users/prashantsingh/protoc3071461866139821779.exe": error=86, Bad CPU type in executable -> [Help 1]


### Steps to reproduce

Try either

`make` 

OR

`make release PROFILES="-Pspark-3.5 -Pscala-2.13"`

### Expected behavior

To pass

### Additional context

```
java --version 
openjdk 17.0.13 2024-10-15 LTS
OpenJDK Runtime Environment Corretto-17.0.13.11.1 (build 17.0.13+11-LTS)
OpenJDK 64-Bit Server VM Corretto-17.0.13.11.1 (build 17.0.13+11-LTS, mixed mode, sharing)
```

--- a/pom.xml
+++ b/pom.xml
@@ -55,7 +55,7 @@ under the License.
     <spark.version>3.4.3</spark.version>
     <spark.version.short>3.4</spark.version.short>
     <spark.maven.scope>provided</spark.maven.scope>
-    <protobuf.version>3.19.6</protobuf.version>
+    <protobuf.version>3.21.12</protobuf.version>
     <parquet.version>1.13.1</parquet.version>
     <parquet.maven.scope>provided</parquet.maven.scope>
     <arrow.version>16.0.0</arrow.version>

